dvc/remote/base.py

```diff
@@ -468,19 +468,20 @@ def download(self, from_info, to_info, name=None, no_progress_bar=False):
             # lets at least show start and finish
             progress.update_target(name, 0, None)
 
-        makedirs(fspath_py35(to_info.parent), exist_ok=True)
+        self.repo.cache.local.makedirs(to_info.parent)
```

Comment:
Still looks wrong. This presumes that `to_info` is in local cache, which might be false. 

We are systematically tangling up cache and remotes, ideally remotes should know nothing about cache.

Reply:
It doesn't presume that. It just uses local cache and its configuration to handle _local_ files.

Reply:
It does presume. If `to_info` is not in cache why do we use it to handle file operations? What effects it will add upon? This is kind of complication that we should avoid.

Reply:
Maybe `chmod` should be moved out of `download()` actually. If this is cache thing then it should be handled in `.pull()` I guess.

Reply:
And `RemoteLOCAL.pull()` may actually use config it owns and not use a hacky trick to employ something elses config.

Reply:
BTW, why are you focusing on download only? Should not you chmod every time you add file to cache, like in `dvc add/run`?

Reply:
In that case there will be a period of time where the cache is there, but it doesn't have proper permissions. Will need to download it to a temporary location first and then move to the cache itself with proper perms.

Reply:
@Suor we do chmod on add/run too.

Reply:
Then you should simply pass `mode` to `.download()`.

Reply:
@efiop I see you rely on `.move()`.